vendors/xilinx/boards/microzed/aws_demos/config_files/FreeRTOSIPConfig.h

```diff
@@ -52,7 +52,7 @@ extern void vLoggingPrintf( const char * pcFormatString,
  * FreeRTOS_netstat() command, and ping replies.  If ipconfigHAS_PRINTF is set to 1
  * then FreeRTOS_printf should be set to the function used to print out the
  * messages. */
-#define ipconfigHAS_PRINTF    1
```

Comment:
Hi, I would not like to switch off all +TCP logging by default.

I understand that the messages about the remaining heap are disturbing ( to me as well ). There are far too many of them.

Many developers ( on the forum ) made grateful use of them: thanks to the logging it became clear that an application was running out of resources.
When people complain about "bugs" in FreeRTOS+TCP, quite often it turns out to be a problem with resources.

Therefore, I propose to change the function to write much less logging about the heap amounts. As an example:
~~~c
#define niPERCENTAGE_100	( 100U )
#define niPERCENTAGE_90		( 90U )

    /* Only write logging if there is a 10% decrease. */
    if( ( uxMinLastSize * niPERCENTAGE_90 ) > ( uxMinSize * niPERCENTAGE_100 ) )
    {
        uxMinLastSize = uxMinSize;
        FreeRTOS_printf( ( "Heap: current %lu lowest %lu\n", xPortGetFreeHeapSize(), uxMinSize ) );
    }
~~~
With this code, there will more logging as the heap has less memory.
When I tested with a Zynq, there was no heap logging at all ( because there is plenty ).

Reply:
Thank you, this change will be undid.